lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const { getOptionValue } = require('internal/options');
+const experimentalImportMetaResolve =
+getOptionValue('--experimental-import-meta-resolve');
+const { PromisePrototypeThen, PromiseReject } = primordials;
+const asyncESM = require('internal/process/esm_loader');
+
+function createImportMetaResolve(defaultParentUrl) {
+  return async function resolve(specifier, parentUrl = defaultParentUrl) {
+    return PromisePrototypeThen(
+      asyncESM.esmLoader.resolve(specifier, parentUrl),
```

Comment:
this is just copy pasted from existing impl, and I do believe so since the hook delegates in that property on the loader instance : https://github.com/nodejs/node/blob/master/lib/internal/modules/esm/loader.js#L470